docs/DeveloperGuide.md

```diff
@@ -504,3 +466,54 @@ testers are expected to do more *exploratory* testing.
+
+The **biggest hurdle** to overcome with this project was **modifying the testing methods**. The test methods were mostly designed for immutable objects and there were strict tests like JsonSerializableAddressBookTest which were the most difficult to pass. This is because the test compares the hardcoded file of clients to a generated file of clients. However, at first since we were making the objects mutable, the generated file becomes different in comparison to the hardcoded file ONLY during the full testing sequence. When the test was done individually, the test passes. We scratched our heads over this for a very long time and eventually managed to solve this problem by changing the equals method for `Client` object and also the `InsurancePlansManager` object.
+
+However, **this fix initially caused other tests break**. This includes tests like `addClaimCommandTest.java` as the `assertCommandSuccess` method in `TestUtil.java` did not compare actual and expected models correctly. This meant that even if there were some changes in the insurance plan manager, sometimes the two models are evaluated as equal to each other. We then had to create new assert methods for commands (like assertInsuranceCommandSuccess) for all the insurance plan related objects.
+
+Our implementations were at **least as difficult compared to half of AB-3 effort**. While we did not have to create many huge structures from scratch like the `ModelManager`, `Logic` and `Storage` components, we had to work with moving parts compared to AB-3. AB-3 does not support any form in automatic sorting. However, since we wanted a quick and easy way for insurance managers to view their most critical claims, we implemented an automatic sorting of claims for each client to show their respective open claims first.
+
+Lastly, as **our project involved money**, which is a sensitive value that has to be precise and accurate. We decided to store all values in cents to avoid having to use float (and having issues with floating point accuracy). However, this led to **greater care required to parse and convert values** to cents and ended up being a very time-consuming method to unit test. This method was still not completely fixed (the constraint of a maximum claim amount of $1 Million was yet to be implemented).
```

Comment:
I think this can be better phrased as: "Lastly, as our project involved **money**, a sensitive value that has to be precise and accurate, we decided to store all values in cents. This was to avoid having to use `float` (and having issues with floating point accuracy)."

Reply:
noted, updated!